packages/ui/origin-ui-v2/libs/ui/utils/src/types/index.ts

```diff
@@ -0,0 +1,332 @@
+  address: string;
+  tokenId: string;
+  deviceId: string;
+  generationFrom: string;
+  generationTo: string;
+}
+
+export type AccountAsset = {
+  asset: IAsset;
+  amount: string;
+};
+
+export type AccountBalance = {
+  available: AccountAsset[];
+  locked: AccountAsset[];
+};
+
+export type ExchangeAccount = {
+  address: string;
+  balances: AccountBalance;
+};
+
+export enum TransferStatus {
+  Unknown,
+  Accepted,
+  Unconfirmed,
+  Confirmed,
+  Error,
+}
+
+export enum TransferDirection {
+  Deposit,
+  Withdrawal,
+}
+
+export interface ITransfer {
+  id: string;
+  userId: string;
+  asset: IAsset;
+  amount: string;
+  transactionHash: string;
+  address: string;
+  status: TransferStatus;
+  confirmationBlock?: number;
+  direction: TransferDirection;
+}
+
+export interface IOrderBookOrderDTO {
+  id: string;
+  price: number;
+  volume: string;
+  product: ProductDTO;
+  userId: string;
+  assetId?: string;
+}
+
+export type TradePriceInfo = {
+  created: string;
+  volume: string;
+  price: number;
+  product: ProductDTO;
+  assetId: string;
+};
+
+export type TOrderBook = {
+  asks: IOrderBookOrderDTO[];
+  bids: IOrderBookOrderDTO[];
+  lastTradedPrice: TradePriceInfo;
+};
+
+export interface IDirectBuyDTO {
+  askId: string;
+  volume: string;
+  price: number;
+}
+
+export interface IOrder {
+  id: string;
+  side: OrderSide;
+  validFrom: string;
+  product: IProductDTO;
+  price: number;
+  startVolume: string;
+  currentVolume: string;
+  directBuyId: string;
+  asset: IAsset;
+  assetId: string;
+  userId: string;
+  filled?: number;
+  demandId?: string;
+}
+
+export interface IDemand {
+  id: string;
+  userId: string;
+  price: string;
+  start: Date;
+  end: Date;
+  volumePerPeriod: string;
+  periodTimeFrame: TimeFrame;
+  product: IProductDTO;
+  bids: Order[];
+  status: DemandStatus;
+}
+
+export type MarketRedirectFilter = {
+  redirectDeviceType: string[];
+  redirectLocation: string[];
+  redirectGridOperator: string[];
+  redirectGenerationFrom: Date;
+  redirectGenerationTo: Date;
+};
+
+export interface IOrder {
+  id: string;
+  side: OrderSide;
+  validFrom: string;
+  product: IProductDTO;
+  price: number;
+  startVolume: string;
+  currentVolume: string;
+  directBuyId: string;
+  asset: IAsset;
+  assetId: string;
+  userId: string;
+  filled?: number;
+}
+
+export type Order = IOrder & { assetId: string; status: OrderStatus };
+
+export type RequestWithdrawalDTO = {
+  readonly assetId: string;
+  readonly address: string;
+  readonly amount: string;
+};
+
+export type BundleItem = {
+  id: string;
+  asset: IAsset;
+  startVolume: BigNumber;
+  currentVolume: BigNumber;
+};
+
+export type Bundle = {
+  id?: string;
+  userId: string;
+  price: number;
+  isCancelled: boolean;
+  items: BundleItem[];
+  volume: BigNumber;
+  own: boolean;
+  splits?: Split[];
+};
+
+export type BundleItemDTO = {
+  assetId: string;
+  volume: string;
+};
+
+export type CreateBundleDTO = {
+  price: number;
```

Comment:
These types will be removed, and the types from appropriate `backend-api-client` package will be used